internal/locald/sandboxmanager/grpc_server.go

```diff
@@ -144,6 +144,10 @@ func (s *grpcServer) registerSandbox(sb *models.Sandbox) {
 	defer s.sbMu.Unlock()
 	sbm, present := s.sbMonitors[sb.Name]
 	if present {
+		// Close all current rev tunnels. When applying the sandbox in the SaaS,
+		// it will update the SDS in the target cluster, causing current tunnels
+		// to disconnect. By closing them before hand, we are more reactive.
+		sbm.closeRevTunnels()
```

Comment:
It is a good observation that an SDS update triggers tunnel closure, and this would help.


But, I think that in light of this TODO https://github.com/signadot/cli/blob/5c8b8cbf11920940a66d468aeab93237e48bde20/internal/locald/sandboxmanager/sandbox_monitor.go#L190 (making updates work), we should not make that harder by closing tunnels here, rather, we should make the updates seamless in case of no changes and close the existing tunnel when there is a change.  I guess this was out of scope of your intended fixes in this PR. 

I have prepared and tested operator and CLI changes which takes the course of action mentioned above and in the TODO, will post shortly.